### ISSUE TITLE
General visualizations issue

### ISSUE THREAD

**Issue by mosheduminer** [sirixdb/sirix-svelte-front-end#251]
There are many visualizations described in @JohannesLichtenberger's master thesis, these need to be recreated.
A PDF of the master thesis is [here](https://github.com/JohannesLichtenberger/master-thesis/raw/master/Master/Thesis/thesis.pdf).

**Comment by villelaitila** [comment]
As a hint, ECharts provides very nice performance as an alternative to D3.js. I really recommend checking echarts.apache.org out. As visualization is not yet deeply implemented into the library, it would be easy to pick the best possible library for the job.

**Comment by mosheduminer** [comment]
@villelaitila thanks for the suggestion! I actually never heard of echarts before, but it looks interesting.
At the moment, I myself am not sure about the details of the visualizations that we need to implement. We definitely want to recreate the visualizations from @JohannesLichtenberger's [thesis](https://github.com/JohannesLichtenberger/master-thesis/blob/master/Master/Thesis/thesis.pdf), and I don't quite understand everything. In particular, I am concerned that we may need to do a lot of custom work, and an out-of-the-box charting library may not be enough.
So, echarts will get us a good deal of the way there, but if I need a custom sunburst implementation, for example, what primitives would echarts provide (if any)?

**Comment by villelaitila** [comment]
I've never done sunburst charts with ECharts but there seems to be a credible set of sunburst examples on the examples page: https://echarts.apache.org/examples/en/index.html

**Comment by mosheduminer** [comment]
Right, I saw those, they use the basic echarts implementation, and should work for #250. The problem is that I'm pretty sure we're going to be implementing visualizations that aren't pure sunbursts, and don't quite fit a classic visualization category (at least to my knowledge).

For example, I can't imagine what Fig. 19 in above mentioned thesis would be called (or how I would implement it): 
![image](https://user-images.githubusercontent.com/47164590/97121162-8be6fd80-16f2-11eb-8c39-cdb8f6590555.png)
But I imagine that I'm going to _need_ to use D3, unless echarts or another library provides easy-to-use primitives.
